src/act.obj.cpp

```diff
@@ -1950,7 +1895,7 @@ int perform_drop(struct char_data * ch, struct obj_data * obj, byte mode,
                     CAP(GET_OBJ_NAME(obj)));
   
   FALSE_CASE_PRINTF(IS_OBJ_STAT(obj, ITEM_EXTRA_KEPT) && !IS_SENATOR(ch), "You'll have to use the KEEP command on %s before you can %s it.", decapitalize_a_an(obj), sname);
-  FALSE_CASE_PRINTF(obj == ch->char_specials.programming, "You can't %s something you are working on.", sname);
+  FALSE_CASE_PRINTF(obj == ch->char_specials.building, "You can't %s something you are working on.", sname);
```

Comment:
I think I don't understand the difference between the `programming` and `building` fields. Why are they separate, and when are they used?

Reply:
.programming refers to the equivalent edit_obj but it's a matrix_file* struct instead of an obj_data* struct. .building is the d->edit_obj equivalent for anything that's an obj_data* struct. For example, creating an OBJ_PART will use .building because it's creating an OBJ. .programming will always result in a struct matrix file which can only exist in storage-enabled devices.